### PR TITLE
feat: add docs for cache management

### PR DESCRIPTION
**Pull request by bietkul**

### CODE REVIEW

File: content/docs/search/reactivesearch-api/reference.md [hunk 1]
```diff
@@ -562,3 +562,6 @@ The following example uses all three functions (`saturation`, `log` and `sigmoid
 
 ### userId
 `String` It allows you to define the user id which will be used to record the Appbase.io analytics.
+
+### useCache
+This property allows you to cache the search requests. The `useCache` property has the higher precedence over the cache preferences. If it is defined in the request body then no matter what are your cache preferences are set, caching would be driven by the `useCache` value.
```
Comment (siddharthlatest) [inline content/docs/search/reactivesearch-api/reference.md]:
```suggestion
`Boolean` This property when set allows you to cache the current search query. The `useCache` property takes precedence irrespective of whether caching is enabled or disabled via the dashboard.
```

File: src/pages/index.js [hunk 2]
```diff
@@ -257,6 +257,21 @@ const HomePage = ({ data, location }) => {
 										],
 									},
 									'7': {
+										title: 'Speed',
```
Comment (siddharthlatest) [inline src/pages/index.js]: @bietkul This should come before Security section.

**Comment by siddharthlatest** [comment]
@bietkul 

- [ ] The icon doesn't match other icons, you can use one that does. The spacing isn't even.



- [ ] Positioning change is needed in the top area as well. Also change `Security Credentials` to `Security`.

File: content/docs/speed/cache-management.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+sidebar: 'docs'
+---
+
+## Overview
+Caching the search requests is helpful to minimize the elasticsearch interactions for frequent queries. It has the following benefits:
+- Superior performance, it can shave up to 100ms of search latencies for many of the expensive searches,
+- Improve the search throughput for the same kinds of queries, as they never need to hit the server.
+
+## Cache Preferences
+![alt cache preferences GUI](https://i.imgur.com/L7P3NYx.png)
```
Comment (siddharthlatest) [inline content/docs/speed/cache-management.md]: @bietkul Use the new image. Also update the image we use for the lock screen in the dashboard.

File: content/docs/speed/cache-management.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+### Enable/Disable Caching
+By default the caching is disabled for clusters, there are two ways to enable request caching:
+1. Set the caching to `true` in cache preferences,
+2. Use the [useCache](/docs/search/reactivesearch-api/reference/#usecache) property as `true` in the [settings](/docs/search/reactivesearch-api/reference/#settings-properties) object of the ReactiveSearch API. It has the higher precedence and if set then cache preferences would be ignored.
+
+### Cache Duration
+The time to live for a cached request in seconds. The default value is 360s. Cache Duration must be in a range of 60s to 3600s(1 hr).
```
Comment (siddharthlatest) [inline content/docs/speed/cache-management.md]: @bietkul The default value is 300s, not 360s.

File: content/docs/speed/cache-management.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+### Memory Limit
+The memory limit for the cached requests in MB(s). The default value is 128MB. It can go up to 4000MB(4GB). Appbase.io uses a cache eviction policy to remove costly requests automatically once it reaches the memory limit.
```
Comment (siddharthlatest) [inline content/docs/speed/cache-management.md]: @bietkul What is the eviction policy based on? Mentioning that is an important detail.

File: content/docs/speed/cache-management.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+### Whitelist indices
+It allows you to define which index/indices should be considered to cache the search queries.
+
+## Clear Cache
+The `Evict Cache` action allows you to free up the memory by deleting cached requests instantly.
+
+## How to identify cached responses?
+Appbase.io sets the `x-request-cache` header as `true` for cached responses. To identify the cached responses you just need to check the presence of `x-request-cache` in the response headers of the request.
```
Comment (siddharthlatest) [inline content/docs/speed/cache-management.md]:
```suggestion
Appbase.io sets the `X-Request-Cache` header as `true` for cached responses, i.e. when caching is enabled and the request was found within the cache.
```

@bietkul You can add a screenshot from Postman for this.

File: content/docs/speed/cache-management.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+### Cache Duration
+The time to live for a cached request in seconds. The default value is 360s. Cache Duration must be in a range of 60s to 3600s(1 hr).
+
+### Memory Limit
+The memory limit for the cached requests in MB(s). The default value is 128MB. It can go up to 4000MB(4GB). Appbase.io uses a cache eviction policy to remove costly requests automatically once it reaches the memory limit.
+
+### Whitelist indices
+It allows you to define which index/indices should be considered to cache the search queries.
+
+## Clear Cache
```
Comment (siddharthlatest) [inline content/docs/speed/cache-management.md]:
```suggestion
## Evict Cache
```

**Review by siddharthlatest — CHANGES_REQUESTED** [review]

File: content/docs/speed/cache-management.md (hunk 5 above)
Comment (siddharthlatest) [reply content/docs/speed/cache-management.md]: @bietkul This isn't addressed.

File: content/docs/speed/cache-management.md (hunk 6 above)
Comment (siddharthlatest) [reply content/docs/speed/cache-management.md]: @bietkul This also isn't addressed.

Comment (siddharthlatest) [reply content/docs/speed/cache-management.md]: @bietkul This also isn't addressed.

File: content/docs/speed/cache-management.md (hunk 5 above)
Comment (bietkul) [reply content/docs/speed/cache-management.md]: @siddharthlatest I've updated the content and included that part into `evict cache` section.
https://i.imgur.com/caZ5mAj.png

File: src/pages/index.js [hunk 8]
```diff
@@ -224,6 +224,21 @@ const HomePage = ({ data, location }) => {
...
+										icon: (
+											<Icon name="clock" className="dropdown-content-icon" />
+										),
+										chapters: [
+											{
+												title: 'Cache Management',
+												link: '/docs/speed/cache-management/',
+												duration: 1,
+												description:
+													'Cache search requests to boost the speed.',
```
Comment (siddharthlatest) [inline src/pages/index.js]:
```suggestion
													'Blazing ⚡️ Elasticsearch search performance with appbase.io cache,
```

File: src/pages/index.js [hunk 9]
```diff
@@ -224,6 +224,21 @@ const HomePage = ({ data, location }) => {
...
+										icon: (
+											<Icon name="clock" className="dropdown-content-icon" />
+										),
+										chapters: [
+											{
+												title: 'Cache Management',
+												link: '/docs/speed/cache-management/',
+												duration: 1,
+												description:
+													'Blazing ⚡️ Elasticsearch search performance with appbase.io cache,
```
Comment (siddharthlatest) [inline src/pages/index.js]:
```suggestion
													'Blazing ⚡️  search performance with appbase.io cache,
```

**Review by siddharthlatest — APPROVED** [review]

**Pull request merged**